app/controllers/courses_controller.rb

```diff
@@ -36,6 +46,26 @@ def show
 
   action_auth_level :manage, :instructor
   def manage
+    if(@course.nil?)
+      flash[:error] = "Cannot manage nil course"
+    end
+
+    if !@course.valid?
+      @course.errors.full_messages.each do |msg|
+        flash[:error] += "<br>#{msg}"
+      end
+    end
+
+    if(@cud.nil?)
+      flash[:error] = "Cannot manage nil course"
+    end
+
+    if !@cud.valid?
```

Comment:
@cud isn't being updated and it's pulled from the db so it's definitely valid here